src/webgpu/shader/execution/expression/call/builtin/atanh.spec.ts

```diff
@@ -41,10 +42,24 @@ g.test('f32')
     };
 
     const cases = [
-      ...biasedRange(-1, -0.9, 20), // discontinuity at x = -1
-      ...biasedRange(1, 0.9, 20), // discontinuity at x = 1
-      ...fullF32Range(),
+      ...biasedRange(kValue.f32.negative.less_than_one, -0.9, 20), // discontinuity at x = -1
+      ...biasedRange(kValue.f32.positive.less_than_one, 0.9, 20), // discontinuity at x = 1
+      ...sourceFilteredF32Range(
+        t.params.inputSource,
+        kValue.f32.negative.less_than_one,
+        kValue.f32.positive.less_than_one
+      ),
     ].map(makeCase);
+
+    // Handle the edge case of -1 and 1 when not doing const-eval
+    if (t.params.inputSource !== 'const') {
+      const edgeCases = [
+        ...biasedRange(-1, kValue.f32.negative.less_than_one, 20),
+        ...biasedRange(1, kValue.f32.positive.less_than_one, 20),
```

Comment:
These should just be -1 and 1. There shouldn't be any f32 values between 1(-1) and positive.less_than_one(negative.less_than_one), because there should only be 1 ULP between them.

 number is a f64 in TS, so it will generate values, but those will get quantized to either or less_than_one when going to f32.

Reply:
done